core/roboconf-target-docker/src/main/java/net/roboconf/target/docker/internal/DockerfileGenerator.java

```diff
@@ -144,6 +148,13 @@ public File generateDockerfile() throws IOException {
 			out.println("COPY rc.local /etc/");
 			out.println("COPY start.sh /usr/local/roboconf-agent/");
 
+			// Copy additional resources in the Karaf deploy directory if needed.
+			// We use the Docker ADD command that accepts remote URLs.
+			if ( this.deployList != null && !this.deployList.isEmpty() ) {
+				for ( final String deploy : this.deployList )
+					out.println("ADD " + deploy + " /usr/local/roboconf-agent/deploy/");
```

Comment:
You're right, it does not work.
Docker `ADD` command does not accept `file:` URLs, and cannot add local files that are not in the build context (like `COPY`). So, the solution seems to copy all `file:` resources in the context directory, and to `ADD` them after. Problem could be clash of resource names, but that's a corner case.